src/dfcx_scrapi/tools/copy_util.py

```diff
@@ -873,147 +919,293 @@ def convert_start_page_dependencies(
+            page_mod.event_handlers = final_ehs
 
+            for transition_route_group in page_mod.transition_route_groups:
+                if transition_route_group in trgs_map.keys():
+                    final_trgs.append(trgs_map[transition_route_group])
+                    print("Found")
+            page_mod.transition_route_groups = final_trgs
+
+        # Change display name to new flow name
+        page_mod.display_name = flow
+
+        return page_mod
 
     def get_page_dependencies(self, obj_list):
         """Pass in DFCX Page object(s) and retrieve all resource dependencies.
-
         Args:
           obj_list: a List of one or more DFCX Page Objects
-
         Returns:
           Dictionary containing all of the resource objects
         """
         resources = defaultdict(list)
-        flow_id = '/'.join(obj_list[0].name.split('/')[0:8])
+        flow_id = "/".join(obj_list[0].name.split("/")[0:8])
 
         # Loop through Pages and find all dependencies
         for page in obj_list:
             resources = self._get_entry_webhooks(page, resources)
             resources = self._get_condition_route_webhooks(page, resources)
             resources = self._get_intent_route_intents(page, resources)
             resources = self._get_form_entity_types(page, resources)
-            resources = self._get_route_groups_and_intents(page, flow_id,
-              resources)
+            resources = self._get_route_groups_and_intents(
+                page, flow_id, resources
+            )
 
         # Start Pages of Flows are special Page-like objects and have different
         # structure, so we need to extract the resources from them differently.
         resources = self._get_flow_intent_route_intents(
-            flow_id, obj_list, resources)
+            flow_id, obj_list, resources
+        )
 
         # Final check to look for additional Entity dependencies
-        if 'intents' in resources:
+        if "intents" in resources:
             resources = self._get_intent_entity_dependencies(resources)
 
         for key in resources:
             resources[key] = set(resources[key])
 
         return resources
+
+    def copy_route_groups(
+        self, agent_id: str, source_flow: str, target_flow: str
+    ):
+        """Creates new route groups in the target flow to match the route groups
+        in the source flow. Requires the pages to exist in the target flow. If
+        the pages do not exist, the destination of the route group routes will
+        be left as empty. Currently only works within an agent, not between
+        different agents. Also note that this will overwrite the content of the
+        route groups in the target flow if they already exist with the same
+        names.
+        """
+        flows_map = self.flows.get_flows_map(agent_id, reverse=True)
+        # Get the route groups for the source flow, and all the maps
+        source_route_groups_map = self.route_groups.get_route_groups_map(
```

Comment:
@SeanScripts can you take a look at variable `source_route_groups_map` ?

The linter says that its unused:
```
************* Module dfcx_scrapi.tools.copy_util
src/dfcx_scrapi/tools/copy_util.py:1136:8: W0612: Unused variable 'source_route_groups_map' (unused-variable)
```

Is it needed? If not, please remove.

cc: @DtStry FYI

Reply:
Yes, thanks, I set that up thinking I would need it but then didn't. Removing it saves an API call.